### PR TITLE
Usage API Idempotency

### PR DESCRIPTION
**Pull request by almalki**
https://github.com/killbill/killbill/issues/502

### CODE REVIEW

File: src/main/java/org/killbill/billing/client/KillBillClient.java [hunk 1]
```diff
@@ -1966,9 +1966,13 @@ private String getResourceFile(final String uri, final String contentType) throw
...
     }
 
-    private void storePluginPropertiesAsParams(final Map<String, String> pluginProperties, final Multimap<String, String> params) {
+    protected void storePluginPropertiesAsParams(final Map<String, String> pluginProperties, final Multimap<String, String> params) {
         for (final String key : pluginProperties.keySet()) {
             params.put(JaxrsResource.QUERY_PLUGIN_PROPERTY, String.format("%s=%s", UTF8UrlEncoder.encode(key), UTF8UrlEncoder.encode(pluginProperties.get(key))));
         }
     }
+
+    protected KillBillHttpClient getKillBillHttpClient(){
```
Comment (pierre) [inline src/main/java/org/killbill/billing/client/KillBillClient.java]: Why these changes?

Comment (almalki) [inline src/main/java/org/killbill/billing/client/KillBillClient.java]: @pierre this was supposed to be another PR, we tried to make KillBillClient extendable to add more methods, this why I added `protected` to make the client and the utility methods accessible to child classes. should I move it to another branch?

Comment (pierre) [inline src/main/java/org/killbill/billing/client/KillBillClient.java]: Moving this would be good, so as not to mix PRs.

We should probably also think it through, i.e. decide which methods should be `protected` (once we do it, they become part of a public API, so we have less flexibility to change them). I'd also be curious to understand the use-case (i.e. why you need it).

**Comment by almalki** [comment]
@pierre fixed

**Comment by pierre** [comment]
Thanks!
